### PR TITLE
Adding tests for Suggestions

### PR DESCRIPTION
**Pull request by tlmii**

### CODE REVIEW

File: src/Microsoft.HttpRepl.Tests/Suggestions/HeaderCompletionTests.cs [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+            Assert.Contains("Upgrade-Insecure-Requests", result, StringComparer.OrdinalIgnoreCase);
+            Assert.Contains("User-Agent", result, StringComparer.OrdinalIgnoreCase);
+        }
+
+        [Fact]
+        public void GetCompletions_NoResults_ReturnsNull()
+        {
+            IEnumerable<string> result = HeaderCompletion.GetCompletions(null, "Not-A-Real-Header");
+
+            Assert.Null(result);
```
Comment (jimmylewis) [inline src/Microsoft.HttpRepl.Tests/Suggestions/HeaderCompletionTests.cs]: spec question: would we rather return (and therefore have to be aware of) null, or would it be better to return an empty collection?

Comment (tlmii) [reply src/Microsoft.HttpRepl.Tests/Suggestions/HeaderCompletionTests.cs]: A quick glance at the code makes it seem like returning an empty collection would be fine as well. The code in both places where this is used is written to handle both scenarios. I think we could simplify it to return an empty collection now that we have the tests to be able to assert that it doesn't return null. Let me take a little deeper look and I'll make a call.

Comment (tlmii) [reply src/Microsoft.HttpRepl.Tests/Suggestions/HeaderCompletionTests.cs]: OK, I changed that up in 425e081.

File: src/Microsoft.HttpRepl/Suggestions/HeaderCompletion.cs [hunk 2]
```diff
@@ -65,36 +65,24 @@ public class HeaderCompletion
...
-                    }
+                    IEnumerable<string> results = programState.GetApplicableContentTypes(method, path);
 
-                    return results.Where(x => !string.IsNullOrEmpty(x) && x.StartsWith(prefix, StringComparison.OrdinalIgnoreCase));
+                    return results?.Where(x => !string.IsNullOrEmpty(x) && x.StartsWith(prefix, StringComparison.OrdinalIgnoreCase));
                 default:
                     return null;
             }
-        }
+        }
```
Comment (jimmylewis) [inline src/Microsoft.HttpRepl/Suggestions/HeaderCompletion.cs]: nit: clean up trailing whitespace

Comment (tlmii) [reply src/Microsoft.HttpRepl/Suggestions/HeaderCompletion.cs]: Done in b2519c5.

**Pull request merged**